### ISSUE TITLE
IterativePSFPhotometry fails with NoOverlapError on _define_fit_data()

### ISSUE THREAD

**Issue by SterlingYM** [astropy/photutils#1769]
I initially commented on  #1589 but it's a closed issue and it doesn't seem to be active, so I'm making a new issue. 

Initial poor fit during the IterativePSFPhotometry leads to an error that looks like ```Initial source at (181, 51) does not overlap with the input data.```. 
This is an issue because a single poorly-fit object can lead to the entire PSF fitting failing, rather than such objects simply being cut or ignored.

A simple cut to remove poorly-fit, out-of-bounds objects like this (perhaps as an option to keep backward compatibility) would be a simple solution, but some people may prefer handling this as an ignored exception rather than cutting data from candidates.
~~~.python
# take results table from the previous iteration
if cut_bad_fits:
    x,y = phot_result['x_fit'],phot_result['y_fit']
    s_edge = (x > 0) & (x < data_shape[1]) & (y > 0) & (y < data_shape[0])
    init_params = QTable()
    init_params['x'] = phot_result['x_fit'].value[s]
    init_params['y'] = phot_result['y_fit'].value[s]
# proceed to the next iteration with the filtered init_params
~~~

Can a fix to this issue be implemented, or am I missing a solution that exists already? Thanks!

(my original post to #1589)
-----------
The same isssue happens in ```PSFPhotometry._define_fit_data()``` when running the code below. I can see the code is written so that it explicitly raises error, rather than ignoring it. 
I can see that this is also due to the poor fit. Is this something that can be ignored, and if so, can this be changed to just warnings?

MWE:
~~~python
import numpy as np
from photutils.datasets import make_test_psf_data, make_noise_image
from photutils.detection import DAOStarFinder
from photutils.psf import IntegratedGaussianPRF, IterativePSFPhotometry, SourceGrouper

### try the code below with nsources=100 and 300
# nsources = 100 # succeeds
nsources = 300 # fails with NoOverlapError

# define psf
sigma_psf = 1.5
psf_model = IntegratedGaussianPRF(flux=1, sigma=sigma_psf)

# generate test data
shape = (150,150)
psf_shape = (11,11)
data, true_params = make_test_psf_data(shape, psf_model, psf_shape,
                                       nsources=nsources, flux_range=(50, 100),
                                       min_separation=1,seed=0)
noise = make_noise_image(data.shape, mean=0, stddev=0.01, seed=0)
data += noise
error = np.abs(noise)

# tools
daofinder = DAOStarFinder(threshold=0.5, fwhm=sigma_psf)
grouper = SourceGrouper(min_separation=3.0*sigma_psf) 

# run photometry
psf_iter = IterativePSFPhotometry(psf_model, fit_shape=(5,5), finder=daofinder,
                                mode='all',grouper=grouper,
                                aperture_radius=3)
phot_result = psf_iter(data, error=error)
~~~

**Comment by larrybradley** [comment]
@SterlingYM Thanks.  I'm currently working on a fix for this issue.

**Comment by larrybradley** [comment]
Thanks again for reporting this.  It was fixed in #1778 and will appear in the next release.
